Userland/Libraries/LibFileSystemAccessClient/Client.h

```diff
@@ -18,20 +19,41 @@
 
 namespace FileSystemAccessClient {
 
+class Client;
+class File {
+public:
+    File(Badge<Client>, NonnullOwnPtr<Core::Stream::File> stream, String filename)
+        : m_stream(move(stream))
+        , m_filename(filename)
+    {
+    }
+
+    Core::Stream::File& stream() const { return *m_stream; }
```

Comment:
The `const` here is quite confusing IMO, can't you return a `const&`?

Reply:
but that would make the most important calls to it (like `read()` and `write()`) inaccessible, right?

Reply:
Right now, this code means "Someone with a const FSAC::File can get the Core::Stream::File from it and call arbitrary methods." That makes sense to me - a const Core::Stream::File would be useless because you can't read or write to it.

Reply:
Why does it make sense to be able to read from a const FSAC::File but not from a const Stream::File?